runtime/lua/vscode-neovim/viewport.lua

```diff
@@ -56,7 +56,12 @@ local function setup_viewport_changed()
 end
 
 function M.setup()
-  setup_viewport_changed()
+  -- Highlighting needs to wait for the viewport-changed event to complete.
+  -- When the UI attaches, there are numerous highlight events (hl_attr_define, grid_line) to process.
+  --
+  -- Without delaying the setup, the viewport-changed event will cause frequent
+  -- pauses in highlight processing, resulting in screen flickering.
+  vim.defer_fn(setup_viewport_changed, 2000)
```

Comment:
Is there anything we can listen for to trigger this, rather than just waiting an arbitrary amount of time?

Reply:
The "viewport-changed" event complements the "win_viewport" event. In most cases, it wouldn't be a significant issue even without the "viewport-changed" event. The delay is more straightforward.